src/Deserialization/Extensions/ValueRefExtensions.cs

```diff
@@ -0,0 +1,39 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using LLVMSharp.Interop;
+
+namespace Symbolica.Deserialization.Extensions
+{
+    internal static class ValueRefExtensions
+    {
+        public static string GetValue(this LLVMValueRef self)
+        {
+            return self.ToString().Split(' ').Last();
```

Comment:
This is the _filth_ that I was talking about when I said I can't see a reliable way to get the "value of a value" without simply dumping and parsing it. If we could somehow find a way to do this we could probably get rid of all that floaty nonsense that went before. Might be worth a fresh set of eyes just in case.

Reply:
I can't see anything better either unfortunately.